### PR TITLE
Update int page

### PR DESCRIPTION
**Pull request by Carlgo11**
I've reworked the regions.rb script to not display sites if:
* They have a regions element
* All of the regions in the regions-element have their own page on 2fa.directory (i.e. "[selectable](https://2fa.directory/api/v3/regions.json)")

That way, people from regions without a dedicated page on 2fa.directory don't see sites only relevant for another region.
i.e. if you're from Luxembourg the banking category isn't filled with U.S. credit unions.

I also replaced `2fa.directory/index.html` with a simple HTML-page linking to the international page. This page should never be displayed to users but if it is shown, we're now more likely to notice that something is wrong with the [redirect script](https://github.com/2factorauth/redirect).

### CODE REVIEW

**Comment by phallobst** [comment]
Unfortunately, this also means that a specific site may not come up in the search results if the user is not on a matching region. At least with `int`, this was guaranteed. I'm afraid this is a step in the wrong direction. (Handling regions like a search term would be way easier IMHO.)

**Comment by Carlgo11** [comment]
Alright I've reverted that change. Now all sites are displayed on the int page regardless of relevance.
I did however do some cleanup of the regions.rb script and the change to `2fa.directory/index.html` is still included.

File: scripts/regions.rb [hunk 1]
```diff
@@ -7,60 +7,46 @@
...
 
 data_dir = './_data'
-websites = JSON.parse(File.read("#{data_dir}/all.json"))
-regions = YAML.load_file("#{data_dir}/regions.yml")
-tmp_dir = '/tmp'
-regions.insert(0, { 'id' => 'int', 'name' => 'global' })
-git_dir = Dir.glob('.git')
-FileUtils.cp_r(git_dir, "#{tmp_dir}/") unless File.exist?("#{tmp_dir}/.git")
+categories = JSON.parse(File.read("#{data_dir}/categories.json"))
+regions = YAML.load_file("#{data_dir}/regions.yml").map { |r| [r['id'], r] }.to_h
```
Comment (ApCoder123) [inline scripts/regions.rb:11]:
```suggestion
regions = YAML.load_file("#{data_dir}/regions.yml").map { |r| [r['id'], { 'name' => r['name'] }] }.to_h
```
For consistency with 'int' key.

File: scripts/regions.rb [hunk 2]
```diff
@@ -7,60 +7,46 @@
...
-    unless website['regions'].nil? || site_regions.empty? || site_regions.include?(region['id']) || region['id'].eql?('int')
-      next
-    end
-
-    next if !site_excluded_regions.nil? && site_excluded_regions.include?(region['id'])
+  JSON.parse(File.read("#{data_dir}/all.json")).each do |name, website|
+    # Ignore if current region is excluded
+    next if website['regions']&.select { |r| r.start_with?('-') }&.map! { |rc| rc.tr('-', '') }&.include?(region['id'])
+    # Ignore unless regions element includes current region or current region is int
+    next unless website['regions']&.reject { |r| r.start_with?('-') }&.include?(key) || key.eql?('int')
```
Comment (ApCoder123) [inline scripts/regions.rb:35]:
```suggestion
    next if website['regions']&.select { |r| r.start_with?('-') }&.map! { |rc| rc.tr('-', '') }&.include?(key)
    # Ignore unless site includes current region, current region is int, site has no regions or site is not excluded in current region
    regions = website['regions']&.reject { |r| r.start_with?('-') }
    next unless regions&.include?(key) || key.eql?('int') || website['regions'].nil? || regions.empty?
```
Current code causes sites with no regions to not be displayed on regional sites and sites with excluded regions to not be displayed on any sites.

File: scripts/regions.rb [hunk 3]
```diff
@@ -7,60 +7,46 @@
...
-    file.write JSON.generate(categories.select { |cat| used_categories[cat['name']] })
+    file.write JSON.generate(categories.select { |cat| used_categories.include? cat['name'] })
   end
 
-  out_dir = "#{Dir.pwd}/_site/#{region['id']}"
-  puts "Building #{region['id']}..."
+  out_dir = "#{Dir.pwd}/_site/#{key}"
+  puts "Building #{key}..."
+  # rubocop:disable Layout/LineLength
   puts `bundle exec jekyll build -s #{dest_dir} --config _config.yml,#{dest_dir}/_config_region.yml -d #{out_dir} --baseurl #{region['id']}`
```
Comment (ApCoder123) [inline scripts/regions.rb:49]:
```suggestion
  puts `bundle exec jekyll build -s #{dest_dir} --config _config.yml,#{dest_dir}/_config_region.yml -d #{out_dir} --baseurl #{key}`
```

**Review by ApCoder123 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**